pkg/api/pluginproxy/access_token_provider.go

```diff
@@ -117,6 +117,10 @@ func (provider *accessTokenProvider) getAccessToken(data templateData) (string,
 	getTokenReq.Header.Add("Content-Type", "application/x-www-form-urlencoded")
 	getTokenReq.Header.Add("Content-Length", strconv.Itoa(len(params.Encode())))
 
+	if err := addAuthHeaders(&getTokenReq.Header, provider.route, data); err != nil {
```

Comment:
Can you please use the existing `addHeaders` function already available in the package?

https://github.com/grafana/grafana/blob/65554269edbeaa08cff46b7c52fc51e96d6eae4d/pkg/api/pluginproxy/ds_auth_provider.go#L108-L118

Reply:
I'm wondering though if each header should be concatenated to the current header value (as [`http.Header.Add`](https://golang.org/pkg/net/http/#Header.Add) does), or if it should _set_ the value with `http.Header.Set`?